src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1788,7 +1789,7 @@ protected void CheckProjectIsNotInCurrentSolution(ProjectId projectId)
         /// </summary>
         protected void CheckProjectHasProjectReference(ProjectId fromProjectId, ProjectReference projectReference)
         {
-            if (!this.CurrentSolution.GetProject(fromProjectId).ProjectReferences.Contains(projectReference))
+            if (!this.CurrentSolution.GetProject(fromProjectId)!.ProjectReferences.Contains(projectReference))
```

Comment:
💭 I do wish there was a better way here

Reply:
Yeah I don't have one but in some sense this seems "right" -- this function is implicitly assuming that the check for the existence of the project has already happened. We could duplicate checks but I'm not sure that's really better.